ui/StatusQ/src/StatusQ/Components/StatusChatList.qml

```diff
@@ -39,248 +39,238 @@ Item {
         model: root.model
         spacing: 0
         interactive: height !== contentHeight
-        section.property: "categoryId"
-        section.criteria: ViewSection.FullString
 
-        section.delegate: Loader {
-            id: statusChatListCategoryItemLoader
-            active: !!section
-
-            required property string section
-
-            sourceComponent: StatusChatListCategoryItem {
-                id: statusChatListCategoryItem
-                
-                function setupPopup() {
-                    categoryPopupMenuSlot.item.categoryId = statusChatListCategoryItemLoader.section
-                }
+        delegate: Loader {
+            id: chatLoader
 
-                function toggleCategory() {
-                    root.model.sourceModel.changeCategoryOpened(statusChatListCategoryItemLoader.section, !opened)
-                    opened = root.model.sourceModel.getCategoryOpenedForCategoryId(statusChatListCategoryItemLoader.section)
+            sourceComponent: {
+                switch(model.type) {
+                    case d.chatTypeCategory: return categoryItemComponent;
```

Comment:
One more question, earlier we had `isCategory` role and it was simple and handy solution. With the type as an enum we need to manually define corresponding enum values on ui side like here in `d.chatTypeCategory`. Wasn't the flag `isCategory` sufficient for that?